daemon/policy.go

```diff
@@ -204,15 +204,13 @@ func (d *Daemon) policyAdd(rules policyAPI.Rules, opts *AddOptions, prefixes []*
 			for _, r := range rules {
 				tmp := d.policy.SearchRLocked(r.Labels)
 				if len(tmp) > 0 {
-					d.dnsPoller.StopPollForDNSName(tmp)
```

Comment:
These `d.dnsPoller.StopPollForDNSName(tmp)` are the ones causing the regenerations.